modules/dlm/src/main/java/org/elasticsearch/dlm/DataLifecycleService.java

```diff
@@ -0,0 +1,321 @@
+                logger.error(() -> format("DLM rollover of [%s] failed", rolloverTarget), e);
+                listener.onFailure(e);
+            }
+        });
+    }
+
+    private void deleteIndex(DeleteIndexRequest deleteIndexRequest, ActionListener<Void> listener) {
+        assert deleteIndexRequest.indices() != null && deleteIndexRequest.indices().length == 1 : "DLM deletes one index at a time";
+        // "saving" the index name here so we don't capture the entire request
+        String targetIndex = deleteIndexRequest.indices()[0];
+        client.admin().indices().delete(deleteIndexRequest, new ActionListener<>() {
+            @Override
+            public void onResponse(AcknowledgedResponse acknowledgedResponse) {
+                listener.onResponse(null);
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                // we're capturin
+                logger.error(() -> format("DLM request to delete [%s] failed", targetIndex), e);
+                listener.onFailure(e);
+            }
+        });
+    }
+
+    @Nullable
+    static TimeValue getRetentionConfiguration(DataStream dataStream) {
+        TimeValue retention = null;
+        if (dataStream.getLifecycle() != null) {
+            retention = dataStream.getLifecycle().getDataRetention();
+        }
+        return retention;
+    }
+
+    /**
+     * Calculate the age of the index since creation or rollover time if the index was already rolled.
+     * The rollover target is the data stream name the index is a part of.
+     */
+    static TimeValue getCreationOrRolloverDate(String rolloverTarget, IndexMetadata index) {
+        RolloverInfo rolloverInfo = index.getRolloverInfos().get(rolloverTarget);
+        if (rolloverInfo != null) {
+            return TimeValue.timeValueMillis(rolloverInfo.getTime());
+        } else {
+            return TimeValue.timeValueMillis(index.getCreationDate());
+        }
+    }
+
+    /**
+     * This is quite a shallow method but the purpose of its existence is to have only one place to modify once we
+     * introduce the index.lifecycle.prefer_ilm setting. Once the prefer_ilm setting exists the method will also
+     * make more sense as it will encapsulate a bit more logic.
+     */
+    private static boolean isManagedByDLM(DataStream parentDataStream, IndexMetadata indexMetadata) {
+        return indexMetadata.getLifecyclePolicyName() == null && parentDataStream.getLifecycle() != null;
+    }
+
+    private RolloverRequest getDefaultRolloverRequest(String dataStream) {
+        RolloverRequest rolloverRequest = new RolloverRequest(dataStream, null).masterNodeTimeout(TimeValue.MAX_VALUE);
+
+        // TODO get rollover from cluster setting once we have it
+        rolloverRequest.addMaxIndexAgeCondition(TimeValue.timeValueDays(30));
+        rolloverRequest.addMaxPrimaryShardSizeCondition(ByteSizeValue.ofGb(50));
+        rolloverRequest.addMaxPrimaryShardDocsCondition(200_000_000);
```

Comment:
I think we need to start a conversation (earlier is better) about what this default should be. I think we should perhaps aim for something a little shorter than 30 days, more in the 7 day range (or shorter if we think we can get away with it).

Reply:
I changed it to 7 days which is what we postulated via the design doc (not sure where my 30 days came from here)
Do you think we should discuss having it lower than 7 days?